File: devtools/agent/bundles/developer/skills/game-dev/ttrpg-gm/package.py
#!/usr/bin/env python3
"""
Simple skill packager - creates .skill file (tar.gz archive)
"""
import tarfile
import os
import sys
from pathlib import Path

def package_skill(skill_path, output_dir=None):
    skill_path = Path(skill_path)
    skill_name = skill_path.name

    if output_dir:
        output_path = Path(output_dir) / f"{skill_name}.skill"
    else:
        output_path = skill_path.parent / f"{skill_name}.skill"

    print(f"Packaging {skill_name}...")

    with tarfile.open(output_path, "w:gz") as tar:
        for item in skill_path.rglob("*"):
            if item.is_file():
                # Skip certain files
                if any(skip in str(item) for skip in ['.git', '__pycache__', '.pyc', 'node_modules']):
                    continue
                arcname = item.relative_to(skill_path.parent)
                tar.add(item, arcname=arcname)
                print(f"  + {item.relative_to(skill_path)}")

    print(f"\n✅ Created: {output_path}")
    print(f"   Size: {output_path.stat().st_size} bytes")
    return output_path

if __name__ == "__main__":
    skill_path = sys.argv[1] if len(sys.argv) > 1 else "/home/moltbot/clawd/skills/ttrpg-gm"
    output_dir = sys.argv[2] if len(sys.argv) > 2 else None

    package_skill(skill_path, output_dir)

File: files/devtools/agent/bundles/developer/skills/game-dev/ttrpg-gm/package.ts
#!/usr/bin/env -S mise x -- bun --install=fallback
import { Crust } from "@crustjs/core@^0.0.19";
import { helpPlugin } from "@crustjs/plugins@^0.1.2";
import { basename, dirname, resolve } from "node:path";

async function packageSkill(sourceArg?: string, outputArg?: string) {
  const source = resolve(sourceArg ?? "/home/moltbot/clawd/skills/ttrpg-gm");
  const output = resolve(outputArg ? `${outputArg}/${basename(source)}.skill` : `${dirname(source)}/${basename(source)}.skill`);
  if (!(await Bun.file(source).exists())) throw new Error(`source directory missing: ${source}`);
  const proc = Bun.spawn(["tar", "-czf", output, "--exclude=.git", "--exclude=__pycache__", "--exclude=node_modules", "-C", dirname(source), basename(source)], { stdout: "pipe", stderr: "pipe" });
  const code = await proc.exited;
  if (code !== 0) throw new Error(await new Response(proc.stderr).text());
  console.log(`Packaging ${basename(source)}...`);
  console.log(`\n✅ Created: ${output}`);
  console.log(`   Size: ${(await Bun.file(output).stat()).size} bytes`);
}
const cli = new Crust("ttrpg-gm").meta({ description: "Package and verify the TTRPG Game Master skill." });
const doctor = cli.sub("doctor").meta({ description: "Report Bun, tar, and skill layout prerequisites." }).run(() => console.log(JSON.stringify({ bun: Bun.version, tar: Boolean(Bun.which("tar")), defaultPath: "/home/moltbot/clawd/skills/ttrpg-gm" }, null, 2)));
const verify = cli.sub("verify").meta({ description: "Verify a skill source directory." }).args([{ name: "source", type: "string", required: false }]).run(async ({ args }) => { const source = resolve(args.source ?? "/home/moltbot/clawd/skills/ttrpg-gm"); if (!(await Bun.file(source).exists())) { process.exitCode = 1; throw new Error(`source directory missing: ${source}`); } console.log(source); });
const pack = cli.sub("package").meta({ description: "Create a .skill archive." }).args([{ name: "source", type: "string", required: false }, { name: "output", type: "string", required: false }]).run(async ({ args }) => packageSkill(args.source, args.output));
cli.use(helpPlugin()).command(doctor).command(verify).command(pack).execute();
